packages/liveblocks-react/src/comments/CommentsRoom.ts

```diff
@@ -93,166 +118,262 @@ export type RoomThreads<TThreadMetadata extends BaseMetadata> =
+    this._cache = value;
+    const event = new CustomEvent("cacheupdate");
+    this.dispatchEvent(event);
   }
 
-  async function revalidateThreads() {
-    pollingHub.threads.pause();
+  /* -------------------------------------------------------------------------------------------------
+   * Event Listener
+   * -----------------------------------------------------------------------------------------------*/
+  addEventListener(
+    key: "cacheupdate",
+    callback: EventListenerOrEventListenerObject | null,
+    options?: boolean | AddEventListenerOptions | undefined
+  ): void;
+
+  addEventListener(
+    key: string,
+    callback: EventListenerOrEventListenerObject | null,
+    options?: boolean | AddEventListenerOptions | undefined
+  ): void {
+    super.addEventListener(key, callback, options);
+  }
 
-    if (numberOfMutations === 0) {
-      if (fetchThreadsPromise === null) {
-        fetchThreadsPromise = room.getThreads();
-      }
-      setThreads(await fetchThreadsPromise);
-      fetchThreadsPromise = null;
-    }
+  removeEventListener(
+    type: "cacheupdate",
+    callback: EventListenerOrEventListenerObject | null,
+    options?: boolean | EventListenerOptions | undefined
+  ): void;
 
-    pollingHub.threads.resume();
+  removeEventListener(
+    type: string,
+    callback: EventListenerOrEventListenerObject | null,
+    options?: boolean | EventListenerOptions | undefined
+  ): void {
+    super.removeEventListener(type, callback, options);
   }
+}
 
-  function subscribe() {
-    if (!unsubscribeRealtimeEvents) {
-      unsubscribeRealtimeEvents = room.events.comments.subscribe(() => {
-        pollingHub.threads.restart(getPollingInterval());
-        void revalidateThreads();
-      });
-    }
+/**
+ * This implementation is inspired by the `swr` library.
+ * Additional modifications were made to adapt it to our specific needs.
+ *
+ * Original `swr` library can be found at [SWR GitHub repository](https://github.com/vercel/swr)
+ */
+export function createCommentsRoom<TThreadMetadata extends BaseMetadata>(
+  room: Room<JsonObject, LsonObject, BaseUserMeta, Json>,
+  errorEventSource: EventSource<CommentsApiError<TThreadMetadata>>
+): CommentsRoom<TThreadMetadata> {
+  const manager = new StateManager<ThreadData<TThreadMetadata>[]>();
 
-    if (!unsubscribeRealtimeConnection) {
-      unsubscribeRealtimeConnection = room.events.status.subscribe((status) => {
-        const nextRealtimeClientConnected = status === "connected";
+  let timestamp = 0;
 
-        if (nextRealtimeClientConnected !== realtimeClientConnected) {
-          realtimeClientConnected = nextRealtimeClientConnected;
-          pollingHub.threads.restart(getPollingInterval());
-        }
-      });
+  let refCount = 0; // Reference count for the number of components with a subscription (via the `subscribe` function)
+  let disposer: (() => void) | undefined; // Disposer function for the `comments` event listener
```

Comment:
I think we should name these differently to be more explicit about what they are/do.